### ISSUE TITLE
TypeDoc reports wrong class as parent class

### ISSUE THREAD

**Issue by benmccann** [TypeStrong/typedoc#1209]
- [X] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Expected Behavior
Should correctly determine parent class.

## Actual Behavior

Says:
>Overrides BubbleController.constructor
Defined in controllers/controller.line.js:27

When it should say
>Overrides DatasetController.constructor
Defined in controllers/controller.line.js:27

Here's the code: https://github.com/chartjs/Chart.js/blob/7397a41fac0e9a57c969148eddc2805c8b249ee7/src/controllers/controller.line.js#L27

## Steps to reproduce the bug

```
git clone git@github.com:chartjs/Chart.js.git
cd Chart.js
git reset --hard 7397a41fac0e9a57c969148eddc2805c8b249ee7
npm install
npx typedoc
```
Now open: 
`file:///directoryWhereYouCheckedOutTheCode/Chart.js/dist/docs/typedoc/classes/_controllers_controller_line_.linecontroller.html`

## Environment
  - Typedoc version: 0.16.9
  - Node.js version: 12.13.1
  - OS: Ubuntu

**Comment by Gerrit0** [comment]
That's... really weird. The hierarchy is correct, so TypeDoc did know what the parent class was, but copied in methods from a child class, which doesn't make sense. 

Looks like this is only set in `mergeDeclarations`, so it must be called inappropriately somewhere. I'll take a closer look next weekend.

https://github.com/TypeStrong/typedoc/blob/d18664d86e85726a725a33e045090cc971d08d65/src/lib/converter/factories/declaration.ts#L214

**Comment by CoreyDotCom** [comment]
This issue still persists in latest TypeDoc... any workarounds??

**Comment by CoreyDotCom** [comment]
cc @Gerrit0 FWIW this was working fine in version 0.15.6 it seems. Can't use that version due to it depending on an older typescript though.

**Comment by Gerrit0** [comment]
So... I've fixed this, for TypeScript projects. Unfortunately, it appears that work done to fix some other issues with inheritance breaks running TypeDoc on Chart.js entirely since we don't have special casing for all of the JSDoc style types. I'm sorry to say that I don't plan on fixing this - at least initially for 0.20. I'd like to support JS projects too, but the primary focus of TypeDoc is TypeScript projects.

For this to be supported:
- Add a special case to convertTypeAlias in src/lib/converter/symbols.ts for JSDocTypedef nodes
- Probably also add a bunch of type converters to src/lib/converter/types.ts for JSDoc type nodes

**Comment by donmccurdy** [comment]
@Gerrit0 I've been tracking this issue because it appears to be affecting my (TypeScript) project. Are you saying that this bug is specific to JS projects? Or that it was affecting all projects, but has since been fixed for TS projects and not JS projects?  If the latter, what version of typedoc should the fix be available in? Thanks!

**Comment by Gerrit0** [comment]
It was affecting all projects, but 0.20 contains a fix for TS projects. It likely also fixes the issue in JS projects, but I was unable to confirm this with the Chart.js test case since it uses JSDoc style types. 0.20 is released as a beta now, and will be released fully before the end of the year.

**Comment by Gerrit0** [comment]
I just double checked, and confirmed that this is fixed in 0.20.9 for Chart.js

**Comment by benmccann** [comment]
That's awesome! Thanks for the amazing 0.20 release and all your help!!
